src/WatchDogs.Infrastructure.DxTrade/DxTradeAuthenticator.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Contracts;
+using Microsoft.Extensions.Options;
+using System.Net.Http.Headers;
+using System.Security.Authentication;
+using System.Text;
+using System.Text.Json;
+
+namespace Infrastructure.DxTrade
+{
+    public class DxTradeAuthenticator : IDxTradeAuthenticator
+    {
+        private const string CookieGroupName = "SET-COOKIE";
+        private const string SessionTokenHeaderName = "JSESSIONID";
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly DxTradeConnectionOptions _connectionOptions;
+        private InMemorySessionTokenStorage _sessionToken;
+
+
+        public DxTradeAuthenticator(IOptions<DxTradeConnectionOptions> configuration, IHttpClientFactory httpClientFactory)
+        {
+            _connectionOptions = configuration.Value;
+            _httpClientFactory = httpClientFactory;
+            _sessionToken = new InMemorySessionTokenStorage();
```

Comment:
Why not inject it via dependency injection?

Reply:
I wasn't sure in this case.